returnn/search.py

```diff
@@ -32,50 +37,43 @@ def __init__(
         search_data,
         model_checkpoint,
         returnn_config,
+        returnn_python_exe,
+        returnn_root,
         *,
         output_mode="py",
         log_verbosity=3,
         device="gpu",
         time_rqmt=4,
         mem_rqmt=4,
         cpu_rqmt=2,
-        returnn_python_exe=None,
-        returnn_root=None,
     ):
         """
         :param dict[str] search_data: dataset used for search
```

Comment:
```suggestion
        :param dict[str,Any] search_data: dataset used for search
```
dict typing takes two arguments. Would we capitalize the d of dict in docstring typing? Would we maybe want to switch to true type annotations if we're on it? (this is really no requirement for my approval)

Reply:
Actually using `typing` objects in the text-style docstrings is not really supported. We should use correct Python typing for new Jobs I guess...